client/container_copy.go

```diff
@@ -84,6 +85,17 @@ func (cli *Client) CopyFromContainer(ctx context.Context, containerID, srcPath s
 	if err != nil {
 		return nil, stat, fmt.Errorf("unable to get resource stat from response: %s", err)
 	}
+	if response.header.Get("Content-Encoding") == "gzip" {
```

Comment:
should also be a new function `readResponsebody(resp serverResponse) (io.ReadCloser, error)`.